plugins/action/action.h

```diff
@@ -119,6 +119,26 @@ class Action
     Result return_to_launch() const;
 
     /**
+     * @brief Send command to transition the drone to fixedwing.
+     *
+     * The associated action will only be executed for VTOL vehicles in multicopter mode.
```

Comment:
Thanks! This will need to be checked of course (once we have got the timeout issue fix). It may be that PX4 would simply ignore the command if sent while already in the current mode (rather than fail) and similarly it might ignore the command for non vtol rather than failing it.

Reply:
@julianoes So based on your change https://github.com/PX4/Firmware/pull/8264 would I be correct in saying that
- on a vtol vehicle the result will either be "success" or timeout, where success is an ack of message received, not that the vehicle will act on it? In other words, if you're already in FW and you call this it will just succeed?
- What will happen on an MC or FW only vehicle - will your added code still get run? If so, then basically this will return success if received by vehicle, which will be a bit odd on MC. Should our code therefore check the type of vehicle we're talking to before even sending the message?

Reply:
> - on a vtol vehicle the result will either be "success" or timeout, where success is an ack of message received, not that the vehicle will act on it? In other words, if you're already in FW and you call this it will just succeed?

correct, it will just succeed.

> - What will happen on an MC or FW only vehicle - will your added code still get run? If so, then basically this will return success if received by vehicle, which will be a bit odd on MC. Should our code therefore check the type of vehicle we're talking to before even sending the message?

It will just time out because the VTOL controller is not running and therefore not responding.

A check to see if it is actually a VTOL and a specific error message for it would be nice!

Reply:
@xgerrmann So
1. Can we check this is VTOL and have a specific error message for other types (rather than getting timeout).
2. So the text needs to change slightly as a  result of Julian's feedback:
    ```
   The associated action will fail with a Result if called for non-VTOL vehicles
   The action is ignored if called when the vehicle is in fixed-wing mode mode.
   ```

@julianoes What about if someone is using firmware pre your fix? Should we also have a PX4 version check?
The problem here is that even in an old version we always want to send the  message, resulting in the timeout. But in an old version we want to ignore the ack result. To me that sounds like we want to have a version of our send that if we don't get an ack on we can force to succeed anyway. THoughts?